### DISCUSSION TITLE
tutorials/3d/3d_rendering_limitations

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#119] [User-contributed notes]
# tutorials/3d/3d_rendering_limitations

Introduction: Due to their focus on performance, real-time rendering engines have many limitations. Godot's renderer is no exception. To work effectively with those limitations, you need to underst...

https://docs.godotengine.org/en/stable/tutorials/3d/3d_rendering_limitations.html

**Comment by eviluess** [comment]
Is there a sample project featuring a basic HDR game/app? So that I can run it to see its final effect on Windows and Android.
I think HDR support is the MOST important feature for a game.

**Reply by Calinou** [reply]
While Godot can internally render in HDR, Godot currently does not support HDR *output*. There is a PR implementing it for Windows: https://github.com/godotengine/godot/pull/94496

**Reply by eviluess** [reply]
What a pity.
What's the bottle-neck for Godot to render output in HDR?
Any nowadays movie player such as mpv can support outputting HDR.

**Reply by Calinou** [reply]
As of 2025, HDR support is being implemented in the following pull requests for a future 4.x release (not 4.5 as it's in feature freeze):

- https://github.com/godotengine/godot/pull/94496
- https://github.com/godotengine/godot/pull/106814
- https://github.com/godotengine/godot/pull/102987
- https://github.com/godotengine/godot/pull/101977
